native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/serializer/protobuf/ProtobufSerializer.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates.
+// Licensed under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//  
+//     http://www.apache.org/licenses/LICENSE-2.0
+//  
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Google.Protobuf;
+
+namespace AWSGsrSerDe.serializer.protobuf
+{
+    /// <summary>
+    /// Avro serialization helper.
+    /// </summary>
+    public class ProtobufSerializer : IDataFormatSerializer
+    {
+        /// <inheritdoc />
+        public byte[] Serialize(object data)
+        {
+            Validate(data);
+            try
+            {
+                var protobufMessage = (IMessage)data;
+                var stream = new MemoryStream();
```

Comment:
Should we be using https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.dispose?view=net-6.0#system-io-stream-dispose ?

Reply:
Good call.